include/openspace/scene/profile.h

```diff
@@ -130,72 +133,41 @@ class Profile {
-    /// functions will be no-ops instead
-    void setIgnoreUpdates(bool ignoreUpdates);
-
-    /// Adds a new asset and checks for duplicates
+    /// Adds a new asset and checks for duplicates unless the `ignoreUpdates` member is
+    /// set to `true`
     void addAsset(const std::string& path);
 
-    /// Removes an asset
+    /// Removes an asset unless the `ignoreUpdates` member is set to `true`
     void removeAsset(const std::string& path);
 
-    /// Removes all assets
-    void clearAssets();
-
-    Version version() const;
-    std::vector<Module> modules() const;
-    std::optional<Meta> meta() const;
-    std::vector<std::string> assets() const;
-    std::vector<Property> properties() const;
-    std::vector<Keybinding> keybindings() const;
-    std::optional<Time> time() const;
-    std::vector<double> deltaTimes() const;
-    std::optional<CameraType> camera() const;
-    std::vector<std::string> markNodes() const;
-    std::vector<std::string> additionalScripts() const;
-
-    void clearMeta();
-    void clearTime();
-    void clearCamera();
-
-    void setVersion(Version v);
-    void setModules(std::vector<Module>& m);
-    void setMeta(Meta m);
-    void setProperties(std::vector<Property>& p);
-    void setKeybindings(std::vector<Keybinding>& k);
-    void setTime(Time t);
-    void setDeltaTimes(std::vector<double> dt);
-    void setCamera(CameraType c);
-    void setMarkNodes(std::vector<std::string>& n);
-    void setAdditionalScripts(std::vector<std::string>& s);
+    static constexpr const Version CurrentVersion = Version{ 1, 1 };
+
+    Version version = CurrentVersion;
+    std::vector<Module> modules;
+    std::optional<Meta> meta;
+    std::vector<std::string> assets;
+    std::vector<Property> properties;
+    std::vector<Action> actions;
+    std::vector<Keybinding> keybindings;
+    std::optional<Time> time;
+    std::vector<double> deltaTimes;
+    std::optional<CameraType> camera;
+    std::vector<std::string> markNodes;
+    std::vector<std::string> additionalScripts;
 
+    bool ignoreUpdates = false;
 
     /**
      * Returns the Lua library that contains all Lua functions available to provide
      * profile functionality.
      * \return The Lua library that contains all Lua functions available for profiles
      */
     static scripting::LuaLibrary luaLibrary();
-
-private:
-    static constexpr const Version CurrentVersion = Version { 1, 0 };
-
-    Version _version = CurrentVersion;
-    std::vector<Module> _modules;
-    std::optional<Meta> _meta;
-    std::vector<std::string> _assets;
-    std::vector<Property> _properties;
-    std::vector<Keybinding> _keybindings;
-    std::optional<Time> _time;
-    std::vector<double> _deltaTimes;
-    std::optional<CameraType> _camera;
-    std::vector<std::string> _markNodes;
-    std::vector<std::string> _additionalScripts;
-
-    bool _ignoreUpdates = false;
 };
 
+std::string convertToScene(const Profile& profile);
```

Comment:
Some comment lines for this function would help to make the conversion more understandable, since this is an important step. Mention that the source file will contain different namespace(s) for previous version(s), etc.

Reply:
Added comments